src/Language/Haskell/TH/DeepStrict.hs

```diff
@@ -361,6 +422,7 @@ isNameDeepStrict typeName args = do
       fmap mconcat . for (zip strictnessReqs args) $ \case
         (Lazy, _)     -> pure DeepStrict
         (Strict, typ) -> isTypeDeepStrict typ []
+        (UnliftedStrictness, typ) -> isTypeDeepUnlifted typ []
```

Comment:
There shouldn't be need to define this whole new `isTypeDeepUnlifted` thing.
Check that it's unlifted with `reifyLevityType fieldType` and then check with `isTypeDeepStrict`

Reply:
surely we need any such type to be unlifted all the way through? or am I misunderstanding how this works?

Reply:
Nah we just want it to be unlifted on the top-level and then deep strict.

For instance:
```haskell
data UBox (a :: Type) :: UnliftedType where
  UBox :: a -> UBox

Array# (UBox Int)
```
is deep strict, even though `Int` is lifted

Reply:
Unlifted basically means this field doesn't need a bang and is still strict

Reply:
alright, I'll simplify down, thanks for the rec